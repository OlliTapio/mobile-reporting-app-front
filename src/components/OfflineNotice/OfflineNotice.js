import React from 'react';
import { StatusBar } from 'react-native';


let statusBar = null;
//for some reason i couldn't get this to work the right way : D
if (this.props.isConnected) {
    statusBar = <StatusBar backgroundColor="#b52424" barStyle="light-content" />;
} else {
    statusBar = <StatusBar backgroundColor="#3d4f7c" barStyle="light-content" />;
}

export default statusBar;

// old version that displayed a red block on top of the creen that said "No Internet Connection"

//import { View, Text, Dimensions, } from 'react-native';
//import styles from "./styles";
//const { width } = Dimensions.get("window");
// const OfflineNotice = () => {
//     return (
//         <View style={[styles.offlineContainer, { width: width }]}>
//             <Text style={styles.offlineText}>No Internet Connection</Text>
//         </View>
//     );
// };
//export default statusBar;
