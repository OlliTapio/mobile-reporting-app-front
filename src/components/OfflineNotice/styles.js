import EStyleSheet from 'react-native-extended-stylesheet';
import {
    StatusBar,
    Platform,
    Dimensions,
} from 'react-native';

const STATUSBAR_HEIGHT = Platform.OS === 'ios' ? 20 : StatusBar.currentHeight;

export default EStyleSheet.create({

    offlineContainer: {
        height: STATUSBAR_HEIGHT,
        width: Dimensions.get('window').width,
        flex: 1,
        //justifyContent: 'center',
        //alignItems: 'center',
        flexDirection: 'row',
        top: 0,
        position: 'absolute',
        alignSelf: 'stretch',
    },

});