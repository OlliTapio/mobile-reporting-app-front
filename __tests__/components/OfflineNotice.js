import 'react-native';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import { OfflineNotice } from '../../src/components/OfflineNotice/OfflineNotice';

configure({ adapter: new Adapter() });

it('renders correctly', () => {
    const tree = renderer.create(
        <OfflineNotice isConnected={true} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
