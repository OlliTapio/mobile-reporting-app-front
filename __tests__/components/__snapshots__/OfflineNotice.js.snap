// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "#3d4f7c",
      },
    ]
  }
/>
`;
